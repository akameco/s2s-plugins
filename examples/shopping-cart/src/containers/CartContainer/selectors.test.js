// @flow
import type { State } from '../../types'
import * as selectros from './selectors'

const state: $Shape<State> = {
  CartContainer: {
    addedIds: [1, 2, 3],
    // $FlowFixMe
    quantityById: { 1: 4, 2: 2, 3: 1 },
  },
  ProductById: {
    // $FlowFixMe
    1: { id: 1, price: 1.99 },
    // $FlowFixMe
    2: { id: 1, price: 4.99 },
    // $FlowFixMe
    3: { id: 1, price: 9.99 },
  },
}

test('getTotal return price total', () => {
  expect(selectros.getTotal(state)).toBe('27.93')
})

test('getCartProducts return products with quantity', () => {
  expect(selectros.getCartProducts(state)).toMatchSnapshot()
})
