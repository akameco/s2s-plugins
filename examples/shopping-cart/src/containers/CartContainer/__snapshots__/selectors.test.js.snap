// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCartProducts return products with quantity 1`] = `
Array [
  Object {
    "id": 1,
    "price": 1.99,
    "quantity": 4,
  },
  Object {
    "id": 1,
    "price": 4.99,
    "quantity": 2,
  },
  Object {
    "id": 1,
    "price": 9.99,
    "quantity": 1,
  },
]
`;
